rasa/nlu/components.py

```diff
@@ -460,347 +466,75 @@ def name(self) -> Text:
-
-        Returns:
-            The list of required package names.
-        """
-        return []
-
-    @classmethod
-    def load(
-        cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Optional["Metadata"] = None,
-        cached_component: Optional["Component"] = None,
-        **kwargs: Any,
-    ) -> "Component":
-        """Loads this component from file.
-
-        After a component has been trained, it will be persisted by
-        calling `persist`. When the pipeline gets loaded again,
-        this component needs to be able to restore itself.
-        Components can rely on any context attributes that are
-        created by :meth:`components.Component.create`
-        calls to components previous to this one.
-
-        Args:
-            meta: Any configuration parameter related to the model.
-            model_dir: The directory to load the component from.
-            model_metadata: The model's :class:`rasa.nlu.model.Metadata`.
-            cached_component: The cached component.
-
-        Returns:
-            the loaded component
-        """
-        if cached_component:
-            return cached_component
-
-        return cls(meta)
-
-    @classmethod
-    def create(
-        cls, component_config: Dict[Text, Any], config: RasaNLUModelConfig
-    ) -> "Component":
-        """Creates this component (e.g. before a training is started).
-
-        Method can access all configuration parameters.
-
-        Args:
-            component_config: The components configuration parameters.
-            config: The model configuration parameters.
-
-        Returns:
-            The created component.
-        """
-        # Check language supporting
-        language = config.language
-        if not cls.can_handle_language(language):
-            # check failed
-            raise UnsupportedLanguageError(cls.name, language)
-
-        return cls(component_config)
-
-    def provide_context(self) -> Optional[Dict[Text, Any]]:
-        """Initializes this component for a new pipeline.
-
-        This function will be called before the training
-        is started and before the first message is processed using
-        the interpreter. The component gets the opportunity to
-        add information to the context that is passed through
-        the pipeline during training and message parsing. Most
-        components do not need to implement this method.
-        It's mostly used to initialize framework environments
-        like MITIE and spacy
-        (e.g. loading word vectors for the pipeline).
-
-        Returns:
-            The updated component configuration.
-        """
-        pass
-
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
+    @abc.abstractmethod
+    def train(self, training_data: TrainingData, **kwargs: Any,) -> None:
```

Comment:
do we make this required? Some components such as `Duckling` or some pretrained featurizers don't need to implement train

Reply:
```suggestion
    def train(self, training_data: TrainingData, **kwargs: Any,) -> Resource:
```